stl/inc/atomic

```diff
@@ -2010,6 +2191,89 @@ template <class _Ty>
 atomic(_Ty) -> atomic<_Ty>;
 #endif // _HAS_CXX17
 
+#if _HAS_CXX20
+template <class _Ty>
+struct atomic_ref : _Choose_atomic_base_t<_Ty, _Ty&> { // atomic reference
+private:
+    using _Base = _Choose_atomic_base_t<_Ty, _Ty&>;
+
+public:
+    // clang-format off
+    static_assert(is_trivially_copyable_v<_Ty> && is_copy_constructible_v<_Ty> && is_move_constructible_v<_Ty>
+        && is_copy_assignable_v<_Ty> && is_move_assignable_v<_Ty>,
+        "atomic_ref<T> requires T to be trivially copyable, copy constructible, move constructible, copy assignable, "
+
+        "and move assignable.");
+    // clang-format on
+
+    using value_type = _Ty;
+
+    explicit atomic_ref(_Ty& _Value) : _Base(_Value) {
+        if constexpr (_Is_potentially_lock_free) {
+            _Check_alignment(_Value);
+        }
+
+        if constexpr (!_Is_potentially_lock_free) {

```

Comment:
I believe this should be `if constexpr` followed by `else` instead of two separate compile-time branches.